drivers/adc/ad7779/ad7779.c

```diff
@@ -451,9 +472,16 @@ int32_t ad7779_get_state(ad7779_dev *dev,
 			 ad7779_ch ch,
 			 ad7779_state *state)
 {
-	*state = dev->state[ch];
+	int32_t ret = 0;
+	uint8_t reg_data;
+
+	if (!dev->read_from_cache) {
+		ret = ad7779_spi_int_reg_read(dev, AD7779_REG_CH_DISABLE, &reg_data);
```

Comment:
missing ret check after the function call.

Reply:
Added ret check here